src/main/java/org/jenkinsci/plugins/xunit/XUnitBuilder.java

```diff
@@ -95,17 +100,17 @@ public ExtraConfiguration getExtraConfiguration() {
     }
 
     @Override
-    public boolean perform(final AbstractBuild<?, ?> build, Launcher launcher, final BuildListener listener)
+    public void perform(final Run<?, ?> build, FilePath workspace, Launcher launcher, final TaskListener listener)
             throws InterruptedException, IOException {
         XUnitProcessor xUnitProcessor = new XUnitProcessor(getTypes(), getThresholds(), getThresholdMode(), getExtraConfiguration());
-        return xUnitProcessor.performXUnit(false, build, listener);
+        xUnitProcessor.performXUnit(false, build, workspace, listener);
     }
 
     public boolean performDryRun(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
             throws InterruptedException, IOException {
         try {
             XUnitProcessor xUnitProcessor = new XUnitProcessor(getTypes(), getThresholds(), getThresholdMode(), getExtraConfiguration());
-            xUnitProcessor.performXUnit(true, build, listener);
+            xUnitProcessor.performXUnit(true, build, build.getWorkspace(), listener);
```

Comment:
`performDryRun` is via https://github.com/jenkinsci/dry-run-lib/blob/master/src/main/java/org/jenkinsci/lib/dryrun/DryRun.java, so had to stay with the existing signature.